hasher-matcher-actioner/hmalib/lambdas/api/submit.py

```diff
@@ -18,6 +18,52 @@
 s3_client = boto3.client("s3")
 dynamodb = boto3.resource("dynamodb")
 
+# Set to 10MB for images
+bottle.BaseRequest.MEMFILE_MAX = 10 * 1024 * 1024
+
+
+def create_presigned_put_url(bucket_name, object_name, file_type, expiration=3600):
+    """
+    Generate a presigned URL to share an S3 object
+    """
+
+    s3_client = boto3.client("s3")
+    try:
+        response = s3_client.generate_presigned_url(
+            "put_object",
+            Params={
+                "Bucket": bucket_name,
+                "Key": object_name,
+                "ContentType": file_type,
+            },
+            ExpiresIn=expiration,
+        )
+    except ClientError as e:
+        logger.error(e)
+        return None
+
+    return response
+
+
+@dataclass
+class InitUploadResponse(JSONifiable):
+    content_id: str
+    file_type: str
+    presigned_url: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+@dataclass
+class InitUploadRequestBody(DictParseable):
```

Comment:
Right, definite at tricky thing to resolve because we want to record the metadata before propagating the content data to the system but it is ambiguous what the system should do if the put operation fails. (important consideration, as you noted, for later/soon)